### ISSUE TITLE
ragas evaluate with llama_index and langchain doesn't seem to work with Azure OpenAI

### ISSUE THREAD

**Issue by Data-drone** [explodinggradients/ragas#114]
Azure OpenAI requires the special parameter `deployment` or `deployment_id`

The langchain wrappers seem to mostly have been updated to accomodate for this but it doesn't seem to work with ragas.

 I ended up getting faithfulness working but updating the `generate` method with:

from 
```

    elif isinstance(llm, BaseChatModel):
        ps = [p.format_messages() for p in prompts]
        result = llm.generate(ps, callbacks=callbacks)
        
        
```
to

```
    elif isinstance(llm, BaseChatModel):
        ps = [p.format_messages() for p in prompts]
        result = llm.generate(ps, callbacks=callbacks, deployment_id='<my_id>', api_version='<my_version>')

```

but with `answer_relevancy` I hit the same issue when it tries to run: 

```
│    91 │   def calculate_similarity(                                                              │
│    92 │   │   self: t.Self, question: str, generated_questions: list[str]                        │
│    93 │   ):                                                                                     │
│ ❱  94 │   │   question_vec = np.asarray(self.embedding.embed_query(question)).reshape(1, -1)     │
│    95 │   │   gen_question_vec = np.asarray(                                                     │
│    96 │   │   │   self.embedding.embed_documents(generated_questions)                            │
│    97 │   │   )
``` 

Any ideas?

**Comment by Data-drone** [comment]
I have setup AzureOpenAI and Llama_index is working fine. But it seems like when you call the service context from ragas it is losing the Azure config somehow?

**Comment by jjmachan** [comment]
We did have a running thread where we fixed this one time but I would love to get on a call with you when you are available and help debug this. The issue is that we haven't gotten access to Azure Openai endpoints so can't debug from our end

the original thread: https://discord.com/channels/1119637219561451644/1123849273562386453/1148224350915407962

it would be great if @Data-drone could commend in that thread, We'll debug this in discord?

**Comment by Data-drone** [comment]
Sure should I try the latest master?

**Comment by jjmachan** [comment]
you could try that but we've released a 0.0.13 version too. Do try it out and let us know :)
